packages/smooth_app/lib/cards/product_cards/question_card.dart

```diff
@@ -35,10 +41,18 @@ class _QuestionCardState extends State<QuestionCard>
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      backgroundColor: const Color(0xff4f4f4f),
-      appBar: AppBar(),
-      body: _buildAnimationSwitcher(),
+    return WillPopScope(
+      onWillPop: () async {
+        if (_lastAnswer != null) {
+          await widget.updateProductUponAnswers();
+        }
+        return true;
+      },
```

Comment:
I cannot picture exactly the whole stack of widgets, but I guess that the product page should await a result and refresh itself according to the result. I'm not a big fan of `WillPopScope`.

Reply:
the drawback of refreshing up the stack is that it will always refresh, sending a callback gives us the flexibility of only refreshing when necessary, so only if an answer has been provided.

The stack is like this ->

Product page sends a callback to refresh the product to Summary card
Summary Card sends a callback to refresh product + questions to question card
Question card refreshes if an answer has been provided

Reply:
I meant something like that
```dart
final bool? refresh = await showQuestion<bool>(...); // also works with a Navigator.push(...) then pop(context, true)
if (refresh ?? false) {
   setState((){});
}
```

Here you have control on whether to refresh or not, from the actual page that needs (or not) to be refreshed.
Like a good old function that returns a value, and a caller that does something different depending on the result.

Maybe my suggestion does not apply in this PR.
Anyway I did approve your PR - I was just asking questions more in a philosophical mode.

Reply:
Ah I didn't know you can do that. I really like how I keep finding new cool things in Flutter. Found this article, https://docs.flutter.dev/cookbook/navigation/returning-data.

I can make the change as you suggested, but we will still need `WillPopScope` to be able to return the `bool? refresh` on pop

Reply:
> I really like how I keep finding new cool things in Flutter

Cool!

> we will still need WillPopScope to be able to return the bool? refresh on pop

Do we? If you just call `Navigator.pop(context, true)` wouldn't that just return `true` as the result of the `await`ed call?
Perhaps in the case when you have no control on the actual return, e.g. when the user presses the back button (and in that case the result would be `null`, which may be good enough in most cases)

Reply:
> > I really like how I keep finding new cool things in Flutter
> 
> Cool!
> 
> > we will still need WillPopScope to be able to return the bool? refresh on pop
> 
> Do we? If you just call `Navigator.pop(context, true)` wouldn't that just return `true` as the result of the `await`ed call? Perhaps in the case when you have no control on the actual return, e.g. when the user presses the back button (and in that case the result would be `null`, which may be good enough in most cases)

Exactly in that case, we will need to intercept with `WillPopScope`. Actually I do this in the gif in this PR, I press back button after answering one question, that already causes the product to be updated with "Poor nutri score"